homeassistant/components/sonos/__init__.py

```diff
@@ -334,22 +338,27 @@ async def _async_ssdp_discovered_player(
             async_dispatcher_send(self.hass, f"{SONOS_VANISHED}-{uid}", reason)
             return
 
-        discovered_ip = urlparse(info.ssdp_location).hostname
-        boot_seqnum = info.ssdp_headers.get("X-RINCON-BOOTSEQ")
         self.async_discovered_player(
             "SSDP",
             info,
-            discovered_ip,
+            cast(str, urlparse(info.ssdp_location).hostname),
             uid,
-            boot_seqnum,
-            info.upnp.get(ssdp.ATTR_UPNP_MODEL_NAME),
+            cast(int, info.ssdp_headers.get("X-RINCON-BOOTSEQ")),
```

Comment:
I think `cast()` does not really perform casting from `str` to `int` but is just a typing helper, right? So should this rather be `int(...)`?

Reply:
Upon further investigation, `__class__` returns it as `str`.